src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -480,9 +720,361 @@ private partial class KeyValueListNode
+            for (int i = 0; i < outBytes.Length; i++)
+                outBytes[i] = HexByteFromChars((char)idData[i * 2], (char)idData[i * 2 + 1]);
+        }
+
+        /// <summary>
+        /// Converts 'idData' which is assumed to be HEX Unicode characters to binary
+        /// puts it in 'outBytes'
+        /// </summary>
+        internal static void SetSpanFromHexChars(Span<byte> outBytes, ReadOnlySpan<char> charData)
+        {
+            Debug.Assert(outBytes.Length * 2 == charData.Length);
+            for (int i = 0; i < outBytes.Length; i++)
+                outBytes[i] = HexByteFromChars(charData[i * 2], charData[i * 2 + 1]);
+        }
+        private static byte HexByteFromChars(char char1, char char2)
+        {
+            return (byte)(HexDigitToBinary(char1) * 16 + HexDigitToBinary(char2));
+        }
+        private static byte HexDigitToBinary(char c)
+        {
+            if ('0' <= c && c <= '9')
+                return (byte)(c - '0');
+            if ('A' <= c && c <= 'F')
+                return (byte)(c - ('A' - 10));
+            if ('a' <= c && c <= 'f')
+                return (byte)(c - ('a' - 10));
+            throw new ArgumentOutOfRangeException("idData");
+        }
+        private static char BinaryToHexDigit(int val)
+        {
+            val &= 0xF;
+            if (val <= 9)
+                return (char)('0' + val);
+            return (char)(('A' - 10) + val);
+        }
+        #endregion
+
+        ulong _id1;
+        ulong _id2;
+        string _asHexString;  // ultimately change to object, and allow several representations based on type.  
+        #endregion
+    }
+
+    /// <summary>
+    /// A SpanId is the format the W3C standard requires for its ID for a single span in a trace.  
+    /// It represents 8 binary bytes of information, typically displayed as 16 characters
+    /// of Hexadecimal.  A SpanId is a STRUCT, and does contain the 8 bytes of binary information
+    /// so there is value in passing it by reference.  It does know how to convert to and
+    /// from its Hexadecimal string representation, tries to avoid changing formats until
+    /// it has to, and caches the string representation after it was created.   
+    /// It is mostly useful as an exchange type.  
+    /// </summary>
+    public unsafe struct ActivitySpanId : IEquatable<ActivitySpanId>
+    {
+        /// <summary>
+        /// Creates a new SpanId from a given set of bytes 'idData'   idData 
+        /// can either be an 8 byte Span of bytes, or a 16 byte Span of Utf8 characters
+        /// and 'isUtf8Chars' indicates which of these it is.   
+        /// </summary>
+        public ActivitySpanId(ReadOnlySpan<byte> idData, bool isUtf8Chars = false)
+        {
+            _id1 = 0;
+            _asHexString = null;
+            fixed (ulong* idPtr = &_id1)
+            {
+                var outBuff = new Span<byte>(idPtr, sizeof(ulong));
```

Comment:
Rather than using `fixed` and pointers like this, can you instead do something like:
```C#
Span<byte> outBuff = MemoryMarshal.AsBytes(MemoryMarshal.CreateSpan(ref _id1, 1));
```
?

Reply:
The System.Diagnostics.DiagnosticsSource is an out of band DLL that can be used on all .NET Platforms (in particular .NET Desktop).   I do need things that work everywhere.    Is the above suggestion still viable given that?

Reply:
> Is the above suggestion still viable given that?

Not as written unfortunately, since MemoryMarshal.CreateSpan is core-only, and there's no OOB-friendly way to create a span that represents an existing field.

If you wanted, you could instead do:
```C#
Span<byte> outBuff = stackalloc byte[sizeof(ulong)];
… // logic that copies to outBuff
_id = MemoryMarshal.Read<ulong>(outBuff);
```